packages/app/src/app/overmind/actions.ts

```diff
@@ -357,3 +367,37 @@ export const openCreateSandboxModal: Action<{ collectionId?: string }> = (
 ) => {
   actions.modals.newSandboxModal.open({ collectionId });
 };
+
+export const validateUsername: AsyncAction<string> = async (
+  { effects, state },
+  userName
+) => {
+  if (!state.pendingUser) return;
+  const validity = await effects.api.validateUsername(userName);
+
+  state.pendingUser.valid = validity.available;
+};
+
+export const finalizeSignUp: AsyncAction<string> = async (
+  { effects, actions, state },
+  username
+) => {
+  if (!state.pendingUser) return;
+  try {
+    await effects.api.finalizeSignUp({
+      id: state.pendingUser.id,
+      username,
+    });
+    window.postMessage(
+      {
+        type: 'signin',
+      },
+      '*'
```

Comment:
Nitpick: to prevent other tabs for listening to the event we can also replace this with "host()" from url generators utilizing.

Reply:
Oh, I mean "protocolAndHost()"!

Reply:
Done :)